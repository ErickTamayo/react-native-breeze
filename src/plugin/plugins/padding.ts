import { PluginFunction, PluginPattern } from "./types";

export type PluginGroups = {
  position?: "x" | "y" | "t" | "r" | "b" | "l" | "s" | "e";
  key: string;
};

export const pattern: PluginPattern = ({ keys }) => {
  // prettier-ignore
  return new RegExp(`^p(?<position>x|y|t|r|b|l|s|e)?-(?<key>${keys("padding")})$`);
};

export const plugin: PluginFunction<PluginGroups> = ({ groups, theme }) => {
  const { position, key } = groups;
  const value = theme<number>(["padding", key]);

  switch (position) {
    case "x":
      return { paddingHorizontal: value };
    case "y":
      return { paddingVertical: value };
    case "t":
      return { paddingTop: value };
    case "r":
      return { paddingRight: value };
    case "b":
      return { paddingBottom: value };
    case "l":
      return { paddingLeft: value };
    case "s":
      return { paddingStart: value };
    case "e":
      return { paddingEnd: value };
    default:
      return { padding: value };
  }
};
