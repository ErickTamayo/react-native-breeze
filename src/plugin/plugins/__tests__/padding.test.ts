import {
  generateInput,
  shouldEvaluateTheCorrectPatternTest,
  shouldMatchOutputSnapshot,
} from "../../helpers/tests";
import { pattern, plugin } from "../padding";

describe("padding", () => {
  const inputs = [
    ...generateInput("p", ["1", "2", "3", "4"]),
    ...generateInput("py", ["1", "2", "3", "4"]),
    ...generateInput("px", ["1", "2", "3", "4"]),
    ...generateInput("pt", ["1", "2", "3", "4"]),
    ...generateInput("pr", ["1", "2", "3", "4"]),
    ...generateInput("pb", ["1", "2", "3", "4"]),
    ...generateInput("pl", ["1", "2", "3", "4"]),
    ...generateInput("ps", ["1", "2", "3", "4"]),
    ...generateInput("pe", ["1", "2", "3", "4"]),
  ];

  shouldEvaluateTheCorrectPatternTest(pattern, {
    shouldMatch: inputs,
    shouldNotMatch: ["unknown", "p-unknown"],
  });

  shouldMatchOutputSnapshot(pattern, plugin, inputs);
});
