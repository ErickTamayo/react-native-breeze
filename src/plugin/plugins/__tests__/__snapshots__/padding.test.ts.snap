// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`padding Should parse (p-1) correctly 1`] = `
Object {
  "padding": 4,
}
`;

exports[`padding Should parse (p-2) correctly 1`] = `
Object {
  "padding": 8,
}
`;

exports[`padding Should parse (p-3) correctly 1`] = `
Object {
  "padding": 12,
}
`;

exports[`padding Should parse (p-4) correctly 1`] = `
Object {
  "padding": 16,
}
`;

exports[`padding Should parse (pb-1) correctly 1`] = `
Object {
  "paddingBottom": 4,
}
`;

exports[`padding Should parse (pb-2) correctly 1`] = `
Object {
  "paddingBottom": 8,
}
`;

exports[`padding Should parse (pb-3) correctly 1`] = `
Object {
  "paddingBottom": 12,
}
`;

exports[`padding Should parse (pb-4) correctly 1`] = `
Object {
  "paddingBottom": 16,
}
`;

exports[`padding Should parse (pe-1) correctly 1`] = `
Object {
  "paddingEnd": 4,
}
`;

exports[`padding Should parse (pe-2) correctly 1`] = `
Object {
  "paddingEnd": 8,
}
`;

exports[`padding Should parse (pe-3) correctly 1`] = `
Object {
  "paddingEnd": 12,
}
`;

exports[`padding Should parse (pe-4) correctly 1`] = `
Object {
  "paddingEnd": 16,
}
`;

exports[`padding Should parse (pl-1) correctly 1`] = `
Object {
  "paddingLeft": 4,
}
`;

exports[`padding Should parse (pl-2) correctly 1`] = `
Object {
  "paddingLeft": 8,
}
`;

exports[`padding Should parse (pl-3) correctly 1`] = `
Object {
  "paddingLeft": 12,
}
`;

exports[`padding Should parse (pl-4) correctly 1`] = `
Object {
  "paddingLeft": 16,
}
`;

exports[`padding Should parse (pr-1) correctly 1`] = `
Object {
  "paddingRight": 4,
}
`;

exports[`padding Should parse (pr-2) correctly 1`] = `
Object {
  "paddingRight": 8,
}
`;

exports[`padding Should parse (pr-3) correctly 1`] = `
Object {
  "paddingRight": 12,
}
`;

exports[`padding Should parse (pr-4) correctly 1`] = `
Object {
  "paddingRight": 16,
}
`;

exports[`padding Should parse (ps-1) correctly 1`] = `
Object {
  "paddingStart": 4,
}
`;

exports[`padding Should parse (ps-2) correctly 1`] = `
Object {
  "paddingStart": 8,
}
`;

exports[`padding Should parse (ps-3) correctly 1`] = `
Object {
  "paddingStart": 12,
}
`;

exports[`padding Should parse (ps-4) correctly 1`] = `
Object {
  "paddingStart": 16,
}
`;

exports[`padding Should parse (pt-1) correctly 1`] = `
Object {
  "paddingTop": 4,
}
`;

exports[`padding Should parse (pt-2) correctly 1`] = `
Object {
  "paddingTop": 8,
}
`;

exports[`padding Should parse (pt-3) correctly 1`] = `
Object {
  "paddingTop": 12,
}
`;

exports[`padding Should parse (pt-4) correctly 1`] = `
Object {
  "paddingTop": 16,
}
`;

exports[`padding Should parse (px-1) correctly 1`] = `
Object {
  "paddingHorizontal": 4,
}
`;

exports[`padding Should parse (px-2) correctly 1`] = `
Object {
  "paddingHorizontal": 8,
}
`;

exports[`padding Should parse (px-3) correctly 1`] = `
Object {
  "paddingHorizontal": 12,
}
`;

exports[`padding Should parse (px-4) correctly 1`] = `
Object {
  "paddingHorizontal": 16,
}
`;

exports[`padding Should parse (py-1) correctly 1`] = `
Object {
  "paddingVertical": 4,
}
`;

exports[`padding Should parse (py-2) correctly 1`] = `
Object {
  "paddingVertical": 8,
}
`;

exports[`padding Should parse (py-3) correctly 1`] = `
Object {
  "paddingVertical": 12,
}
`;

exports[`padding Should parse (py-4) correctly 1`] = `
Object {
  "paddingVertical": 16,
}
`;
